### PR TITLE
kube-aggregator: sync startup with APIService handlers

### PR DESCRIPTION
**Pull request by sttts**
There is a race of the apiserver setting up the api service handlers and reporting readiness. This PR adds a synchronization between two via a post-start hook.

/kind bug

```release-note
NONE
```

### CODE REVIEW

**Comment by lavalamp** [comment]
Some sort of a test would be nice?

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -263,6 +271,34 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
...
+			expected := sets.NewString()
+			for _, apiservice := range apiservices {
+				if v1helper.IsAPIServiceConditionTrue(apiservice, v1.Available) {
+					expected.Insert(apiservice.Name)
+				}
+			}
+
+			s.proxyHandlersLock.Lock()
+			handled := sets.StringKeySet(s.proxyHandlers)
+			s.proxyHandlersLock.Unlock()
```
Comment (yue9944882) [inline staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:291]: nit: defer?

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go [hunk 2]
```diff
@@ -331,9 +370,16 @@ func (s *APIAggregator) AddAPIService(apiService *v1.APIService) error {
 	}
 	proxyHandler.updateAPIService(apiService)
 	if s.openAPIAggregationController != nil {
-		s.openAPIAggregationController.AddAPIService(proxyHandler, apiService)
+		// this is calling a controller.  It should already handle being async.
+		go func() {
```
Comment (yue9944882) [inline staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:374]: this is suspicious. the `AddAPIService` is merely pushing tasks into queue, we don't need to spawn a new routine, is it?

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go [hunk 3]
```diff
@@ -18,11 +18,18 @@ package apiserver
 
 import (
 	"fmt"
+	"k8s.io/klog/v2"
```
Comment (marun) [inline staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:21]: Import ordering?

**Pull request closed without merging**